src/Core/Form/ChoiceProvider/LanguageChoiceProvider.php

```diff
@@ -38,16 +39,16 @@
 final class LanguageChoiceProvider implements FormChoiceProviderInterface
 {
     /**
-     * @var array
+     * @var LanguageDataProvider
      */
-    private $languages;
+    private $languageDataProvider;
 
     /**
-     * @param array $languages
+     * @param LanguageDataProvider $languageDataProvider
      */
-    public function __construct(array $languages)
+    public function __construct(LanguageDataProvider $languageDataProvider)
```

Comment:
you are relying on service from Adapter namespace, maybe it have interface to rely on?

Reply:
if its out of scope then I think you should leave the array $languages here until we introduce some kind of provider in core.